src/model/wallet.ts

```diff
@@ -268,7 +261,7 @@ export const loadWallet = async <S extends Screen>({
 }: {
   address?: EthereumAddress;
   showErrorIfNotLoaded?: boolean;
-  provider?: Provider;
+  provider: Provider;
```

Comment:
a lot of changes in this file are also related to making `provider` arg non-optional; please pay attention to where these functions are being called and that they're passing in valid `provider` objects